src/host/thread_host.hpp

```diff
@@ -113,7 +114,7 @@ enum ThreadEnabledState
  *
  * The APIs are unified for both NCP and RCP cases.
  */
-class ThreadHost : virtual public NetworkProperties
+class ThreadHost : virtual public NetworkProperties, public UdpProxy::Dependencies
```

Comment:
Does `UdpProxy::Dependencies` need to be a base class of `ThreadHost`?

Reply:
Semantically I think `UdpProxy::Dependencies` may be considered as a platform-specific dependency while `ThreadHost` sounds like a set of user-facing APIs. 

This difference makes me feel it inappropriate for `ThreadHost` to inherit `UdpProxy::Dependencies`.

However, this is not a major problem for now. When there are more classes we'll have more clarity about how to refactor it. I'm fine with it as-is.